sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/util/tracing/LoggingTracerProvider.java

```diff
@@ -0,0 +1,190 @@
+        return new LoggingTracer(options instanceof LoggingTracingOptions ? options : DEFAULT_OPTIONS);
+    }
+
+    /**
+     * Options to configure {@link LoggingTracerProvider}.
+     */
+    public static class LoggingTracingOptions extends TracingOptions {
+        /**
+         * Creates new instance of {@link LoggingTracingOptions}
+         */
+        public LoggingTracingOptions() {
+            super(LoggingTracerProvider.class);
+        }
+    }
+
+    /**
+     * Logging tracer implementation.
+     */
+    private static class LoggingTracer implements Tracer {
+        private final boolean isEnabled;
+        LoggingTracer(TracingOptions options) {
+            this.isEnabled = options.isEnabled();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return isEnabled;
+        }
+
+        @Override
+        public Context start(String name, Context context) {
+            if (!isEnabled) {
+                return context;
+            }
+            LoggingSpan span = new LoggingSpan(name, SpanKind.INTERNAL, getSpan(context));
+            return context.addData("span", span);
+        }
+
+        @Override
+        public Context start(String name, StartSpanOptions options, Context context) {
+            if (!isEnabled) {
+                return context;
+            }
+
+            LoggingSpan span = new LoggingSpan(name, options.getSpanKind(), getSpan(context));
+            options.getAttributes().forEach((k, v) -> span.addKeyValue(k, v));
+            span.addKeyValue("startTimestamp", options.getStartTimestamp());
+            return context.addData("span", span);
+        }
+
+        @Override
+        public void end(String errorMessage, Throwable throwable, Context context) {
+            if (isEnabled) {
+                LoggingSpan span = getSpan(context);
+                span.addKeyValue("status", errorMessage);
+                span.end(throwable);
+            }
+        }
+
+        @Override
+        public void setAttribute(String key, String value, Context context) {
+            getSpan(context).addKeyValue(key, value);
+        }
+
+        @Override
+        public void setAttribute(String key, long value, Context context) {
+            getSpan(context).addKeyValue(key, value);
+        }
+
+        @Override
+        public void injectContext(BiConsumer<String, String> headerSetter, Context context) {
+            LoggingSpan span = getSpan(context);
+            headerSetter.accept("traceparent", String.format("00-%s-%s-01", span.getTraceId(), span.getSpanId()));
+        }
+
+        @Override
+        public Context extractContext(Function<String, String> headerGetter) {
+            String traceparent = headerGetter.apply("traceparent");
+            if (traceparent == null) {
+                return Context.NONE;
+            }
+
+            String traceId = traceparent.substring(3, 35);
+            String spanId = traceparent.substring(36, 52);
```

Comment:
These numbers seem pretty magical

Reply:
added comment - it's coming from W3C trace-context